python/cudf/cudf/core/column/decimal.py

```diff
@@ -357,12 +357,23 @@ def _get_decimal_type(lhs_dtype, rhs_dtype, op):
     if op in {"__add__", "__sub__"}:
         scale = max(s1, s2)
         precision = scale + max(p1 - s1, p2 - s2) + 1
-    elif op == "__mul__":
-        scale = s1 + s2
-        precision = p1 + p2 + 1
-    elif op == "__div__":
-        scale = max(6, s1 + p2 + 1)
-        precision = p1 - s1 + s2 + scale
+        if precision > Decimal128Dtype.MAX_PRECISION:
+            precision = Decimal128Dtype.MAX_PRECISION
+            scale = Decimal128Dtype.MAX_PRECISION - max(p1 - s1, p2 - s2)
+    elif op in {"__mul__", "__div__"}:
+        if op == "__mul__":
+            scale = s1 + s2
+            precision = p1 + p2 + 1
+        else:
+            scale = max(6, s1 + p2 + 1)
+            precision = p1 - s1 + s2 + scale
```

Comment:
cc @brandon-b-miller not a fan of the nested conditionals here, did this so we wouldn't have to repeat the scale/precision reduction code below for both the `__mul__` and `__div__` - any preference here?

Reply:
spent a little time trying to organize it differently and concluded this is about as good as it gets.